src/main/java/net/dries007/tfc/common/blocks/devices/IngotPileBlock.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.blocks.devices;
+
+import java.util.Random;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.IntegerProperty;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+import net.minecraftforge.items.ItemHandlerHelper;
+
+import net.dries007.tfc.common.blockentities.TFCBlockEntities;
+import net.dries007.tfc.common.blocks.EntityBlockExtension;
+import net.dries007.tfc.common.blocks.ExtendedBlock;
+import net.dries007.tfc.common.blocks.ExtendedProperties;
+import net.dries007.tfc.common.blocks.TFCBlockStateProperties;
+import net.dries007.tfc.util.Helpers;
+
+public class IngotPileBlock extends ExtendedBlock implements EntityBlockExtension
+{
+    public static final IntegerProperty COUNT = TFCBlockStateProperties.COUNT_1_64;
+
+    private static final VoxelShape[] SHAPES = {
+        box(0, 0, 0, 16, 2, 16),
+        box(0, 0, 0, 16, 4, 16),
+        box(0, 0, 0, 16, 6, 16),
+        box(0, 0, 0, 16, 8, 16),
+        box(0, 0, 0, 16, 10, 16),
+        box(0, 0, 0, 16, 12, 16),
+        box(0, 0, 0, 16, 14, 16),
+        Shapes.block()
+    };
+
+    public IngotPileBlock(ExtendedProperties properties)
+    {
+        super(properties);
+
+        registerDefaultState(getStateDefinition().any().setValue(COUNT, 1));
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public BlockState updateShape(BlockState state, Direction direction, BlockState neighborState, LevelAccessor level, BlockPos currentPos, BlockPos neighborPos)
+    {
+        if (direction == Direction.DOWN && !neighborState.isFaceSturdy(level, neighborPos, direction.getOpposite()) && !Helpers.isBlock(neighborState, this))
+        {
+            level.scheduleTick(currentPos, this, 0);
```

Comment:
does 0 tick delay work? not aware of its use in vanilla

Reply:
This is interesting, because
- iirc it did work (I'll test again to make sure)
- I can't find anywhere in vanilla that does 0 ticks
- I can find other places in TFC that do 0 ticks.
... which is interesting.